src/client/dfuse/dfuse_main.c

```diff
@@ -174,40 +174,40 @@ main(int argc, char **argv)
 
 	if (!dfuse_info->di_mountpoint) {
 		DFUSE_LOG_ERROR("Mountpoint is required");
-		D_GOTO(out, ret = -DER_INVAL);
+		D_GOTO(out_free, ret = -DER_INVAL);
 	}
 
 	/* Is this required, or can we assume some kind of default for
 	 * this.
 	 */
 	if (!svcl) {
 		DFUSE_LOG_ERROR("Svcl is required");
-		D_GOTO(out, ret = -DER_INVAL);
+		D_GOTO(out_free, ret = -DER_INVAL);
 	}
 
 	DFUSE_TRA_ROOT(dfuse_info, "dfuse_info");
 
 	if (dfuse_info->di_pool &&
 	    (uuid_parse(dfuse_info->di_pool, pool_uuid) < 0)) {
 		DFUSE_LOG_ERROR("Invalid pool uuid");
-		D_GOTO(out, ret = -DER_INVAL);
+		D_GOTO(out_free, ret = -DER_INVAL);
 	}
 
 	if (dfuse_info->di_cont &&
 	    (uuid_parse(dfuse_info->di_cont, co_uuid) < 0)) {
 		DFUSE_LOG_ERROR("Invalid container uuid");
-		D_GOTO(out, ret = -DER_INVAL);
+		D_GOTO(out_free, ret = -DER_INVAL);
 	}
 
 	dfuse_info->di_svcl = daos_rank_list_parse(svcl, ":");
```

Comment:
This is where dfuse_info->di_svcl gets set so lines above this should use out, and lines below it should use out_free?

It looks like this change would leak dfuse_info in a lot of cases.

Reply:
Hmmm, sorry Ashley, but I don't understand ...

Reply:
Above this line all gotos on error cases will attempt to free the rank list where it's not initialized yet. So this could still cause issues.

Reply:
Well d_rank_list_free() checks if list pointer is NULL as it must be for a malloc()'ed and uninitialised one .... So this should work, with the minimum changes I have pushed.